### ISSUE TITLE
Postgres Flexible server restore does not restore from WAL logs

### ISSUE THREAD

**Issue by yadavama-intel** [Azure/azure-cli#29772]
### Describe the bug

Encountered a problem while restoring from a backup that resides in another resource-group.
Backup snapshot is taken at 11 pm every day.
Restore operation triggered with restore-time 22nd Aug 10 AM but data only shown till 21st Aug 11 pm, there were some transactions happened after 21st Aug 11 pm but they are not visible in restored server. What am i missing here?

According to [this](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-backup-restore#geo-redundant-backup-and-restore) document - "The database will be recovered until 10:00 AM of August 22 by using the transaction log backup from August 21, 11:00 PM, to August 22, 10:00 AM." But unfortunately i do not see the latest data

### Related command

az postgres flexible-server restore --resource-group rg-dev-2 --name pgsql-dev-2-restore
--source-server /subscriptions/{my-sub-id}/resourceGroups/rg-dev-1/providers/Microsoft.DBforPostgreSQL/flexibleServers/pgsql-rg-dev-1

### Errors

No errors, but there is data discrepancy in tables.
Entities created after snapshot backup and restore operation is not visible in restored pgsql server.

### Issue script & Debug output
Command -

`az postgres flexible-server restore --resource-group rg-dev-2 --name pgsql-dev-2-restore
--source-server /subscriptions/{my-sub-id}/resourceGroups/rg-dev-1/providers/Microsoft.DBforPostgreSQL/flexibleServers/pgsql-rg-dev-1
`

Output - 

`{
  "administratorLogin": "tpeadmin",
  "administratorLoginPassword": null,
  "authConfig": {
    "activeDirectoryAuth": "Enabled",
    "passwordAuth": "Enabled",
    "tenantId": "46c98d88-e344-4ed4-8496-4ed7712e255d"
  },
  "availabilityZone": "3",
  "backup": {
    "backupRetentionDays": 7,
    "earliestRestoreDate": "2024-08-22T05:13:23.359191+00:00",
    "geoRedundantBackup": "Disabled"
  },
  "createMode": null,
  "dataEncryption": {
    "geoBackupEncryptionKeyStatus": null,
    "geoBackupKeyUri": null,
    "geoBackupUserAssignedIdentityId": null,
    "primaryEncryptionKeyStatus": null,
    "primaryKeyUri": null,
    "primaryUserAssignedIdentityId": null,
    "type": "SystemManaged"
  },
  "fullyQualifiedDomainName": "pgsql-laas-restore-backup-22aug-latest.postgres.database.azure.com",
  "highAvailability": {
    "mode": "Disabled",
    "standbyAvailabilityZone": null,
    "state": "NotEnabled"
  },
  "id": "/subscriptions/b271e3b3-f726-44fb-9c88-32e52f7f06aa/resourceGroups/rg-laas-restore/providers/Microsoft.DBforPostgreSQL/flexibleServers/pgsql-laas-restore-backup-22aug-latest",
  "identity": null,
  "location": "East US",
  "maintenanceWindow": {
    "customWindow": "Disabled",
    "dayOfWeek": 0,
    "startHour": 0,
    "startMinute": 0
  },
  "minorVersion": "12",
  "name": "pgsql-laas-restore-backup-22aug-latest",
  "network": {
    "delegatedSubnetResourceId": "/subscriptions/b271e3b3-f726-44fb-9c88-32e52f7f06aa/resourceGroups/rg-laas-restore/providers/Microsoft.Network/virtualNetworks/laasVnet/subnets/laasPgSqlSubnet",
    "privateDnsZoneArmResourceId": "/subscriptions/b271e3b3-f726-44fb-9c88-32e52f7f06aa/resourceGroups/rg-laas-restore/providers/Microsoft.Network/privateDnsZones/pgsql-laas-restore.private.postgres.database.azure.com",
    "publicNetworkAccess": "Disabled"
  },
  "pointInTimeUtc": null,
  "privateEndpointConnections": null,
  "replica": {
    "capacity": 5,
    "promoteMode": null,
    "promoteOption": null,
    "replicationState": null,
    "role": "Primary"
  },
  "replicaCapacity": 5,
  "replicationRole": "Primary",
  "resourceGroup": "rg-laas-restore",
  "sku": {
    "name": "Standard_D4ds_v4",
    "tier": "GeneralPurpose"
  },
  "sourceServerResourceId": null,
  "state": "Ready",
  "storage": {
    "autoGrow": "Disabled",
    "iops": 500,
    "storageSizeGb": 128,
    "throughput": null,
    "tier": "P10",
    "type": ""
  },
  "systemData": {
    "createdAt": "2024-08-22T05:13:23.359191+00:00",
    "createdBy": null,
    "createdByType": null,
    "lastModifiedAt": null,
    "lastModifiedBy": null,
    "lastModifiedByType": null
  },
  "tags": null,
  "type": "Microsoft.DBforPostgreSQL/flexibleServers",
  "version": "14"
}`

### Expected behavior

Eventually the data should be consistent between source server and destination (restored) server.

### Environment Summary

azure-cli :  2.61.0
postgres : Azure postgres flexible server

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by nachoalonsoportillo** [comment]
@yadavama-intel Thanks for having created this issue. As we discovered, the cause of this issue was due to the way you were passing the value to the `--restore-time` parameter.

Value is expected to be in ISO8601 format https://learn.microsoft.com/en-us/cli/azure/postgres/flexible-server?view=azure-cli-latest#az-postgres-flexible-server-restore-optional-parameters, hence time has to be expressed in 24-hour clock.
